### PR TITLE
Added Sample and UT for NFC adapter class

### PR DESCRIPTION
**Pull request by anil-shukla-axway**
https://jira.appcelerator.org/browse/MOD-2843

### CODE REVIEW

**Comment by anil-shukla-axway** [comment]
@fasihul-axway  @satinder-singh-axway  Please review this PR.

File: test/unit/specs/module.spec.js [hunk 1]
```diff
@@ -19,6 +19,75 @@ describe('ti.nfc', () => {
 			it('.moduleId', () => {
 				expect(NFC.moduleId).toBe('ti.nfc');
 			});
+
+			describe('methods', () => {
+				it('nfcAdapter should be defined', () => {
+					const nfcAdapter = NFC.createNfcAdapter();
```
Comment (fasihul-axway) [inline test/unit/specs/module.spec.js:25]: Its good to create a single instance of Adapter and reuse it everywhere.

**Review by fasihul-axway — COMMENTED** [review]
Please see my comment, few minor adjustment required before merging this PR.

File: ios/Classes/TiNfcNfcAdapterProxy.m [hunk 2]
```diff
@@ -32,26 +39,110 @@ - (NFCNDEFReaderSession *)nfcSession
...
+    _nfcTagReadersession = nil;
+  }
 }
 
-- (void)invalidate:(id)unused
+- (TiNfcMiFareUltralightTagTechnology *)createTagTechMifareUltralight:(id)args
 {
-  [[self nfcSession] invalidateSession];
-  _nfcSession = nil;
+  if ([args valueForKey:@"tag"] == nil) {
```
[the remark this answers is not in the document]
Comment (anil-shukla-axway) [reply ios/Classes/TiNfcNfcAdapterProxy.m]: resolved

File: ios/example/app.js [hunk 3]
```diff
@@ -61,12 +62,24 @@ var btn = Ti.UI.createButton({
...
+    if (!nfcAdapter.isEnabled(sessionType)) {
     Ti.API.error('This device does not support NFC capabilities!');
     return;
   }
+  nfcAdapter.begin(sessionType); // This is required for iOS only. Use "invalidate()" to invalidate a session.
+});
 
-  nfcAdapter.begin(); // This is required for iOS only. Use "invalidate()" to invalidate a session.
+nfcAdapter.addEventListener('didDetectTags', function (e) {
+    var mifare = nfcAdapter.createTagTechMifareUltralight({'tag':e.tags[0]});
```
[the remark this answers is not in the document]
Comment (anil-shukla-axway) [reply ios/example/app.js:73]: As I am sending only one polling option, I am only getting one tag object in array and I am fetching it like this.

File: test/unit/specs/module.spec.js (hunk 1 above)
Comment (anil-shukla-axway) [reply test/unit/specs/module.spec.js:25]: If i am defining the same any where else, it's giving error of undefined object. That is why defined separately in each 'it' block.

**Review by vikas-goyal-axway — APPROVED** [review]

**Review by fasihul-axway — APPROVED** [review]

**Pull request merged**